File: app/src/main/java/org/dhis2/usescases/splash/SplashModule.java
package org.dhis2.usescases.splash;

import com.squareup.sqlbrite2.BriteDatabase;

import org.dhis2.data.dagger.PerActivity;
import org.dhis2.data.server.ServerComponent;
import org.dhis2.data.server.UserManager;

import androidx.annotation.Nullable;
import dagger.Module;
import dagger.Provides;

/**
 * QUADRAM. Created by ppajuelo on 07/02/2018.
 */

@Module
public final class SplashModule {

    private final UserManager userManager;

    SplashModule(@Nullable ServerComponent serverComponent) {
        this.userManager = serverComponent == null ? null : serverComponent.userManager();
    }

    @Provides
    @PerActivity
    SplashContracts.SplashPresenter providePresenter(SplashRepository splashRepository) {
        return new SplashPresenterImpl(userManager, splashRepository);
    }

    @Provides
    @PerActivity
    SplashRepository splashRepository(BriteDatabase briteDatabase) {
        return new SplashRepositoryImpl(briteDatabase);
    }

}

File: app/src/main/java/org/dhis2/usescases/splash/SplashModule.kt
package org.dhis2.usescases.splash

import dagger.Module
import dagger.Provides
import javax.inject.Named
import org.dhis2.commons.di.dagger.PerActivity
import org.dhis2.commons.prefs.PreferenceProvider
import org.dhis2.commons.reporting.CrashReportController
import org.dhis2.commons.schedulers.SchedulerProvider
import org.dhis2.data.server.ServerComponent
import org.dhis2.data.server.UserManager
import org.dhis2.usescases.splash.SplashActivity.Companion.FLAG

/**
 * QUADRAM. Created by ppajuelo on 07/02/2018.
 */

@Module
class SplashModule internal constructor(
    private val splashView: SplashView,
    serverComponent: ServerComponent?
) {

    private val userManager: UserManager? = serverComponent?.userManager()

    @Provides
    @PerActivity
    fun providePresenter(
        schedulerProvider: SchedulerProvider,
        preferenceProvider: PreferenceProvider,
        crashReportController: CrashReportController
    ): SplashPresenter {
        return SplashPresenter(
            splashView,
            userManager,
            schedulerProvider,
            preferenceProvider,
            crashReportController
        )
    }

    @Provides
    @PerActivity
    @Named(FLAG)
    fun provideFlag(): String {
        return if (userManager?.d2 != null && userManager.isUserLoggedIn.blockingFirst()) {
            val systemSetting =
                userManager.d2.systemSettingModule().systemSetting().flag().blockingGet()
            if (systemSetting != null) {
                systemSetting.value() ?: ""
            } else {
                ""
            }
        } else {
            ""
        }
    }
}
